src/engine/shared/config_variables.h

```diff
@@ -121,6 +121,8 @@ MACRO_CONFIG_INT(InfEnableNinja, inf_enable_ninja, 1, 0, 1, CFGFLAG_SERVER, "Mak
 MACRO_CONFIG_INT(InfEnableMedic, inf_enable_medic, 1, 0, 1, CFGFLAG_SERVER, "Makes the medic class available")
 MACRO_CONFIG_INT(InfEnableHero, inf_enable_hero, 1, 0, 1, CFGFLAG_SERVER, "Makes the hero class available")
 
+MACRO_CONFIG_INT(InfMinPlayersForEngineer, inf_min_players_for_engineer, 0, 0, 100, CFGFLAG_SERVER, "Minimum number of players that are needed to enable Engineer class")
```

Comment:
this is awesome!

Reply:
This doesn't prevent Engineers from picked by random choice.
There is a lot of duplication, I'm working on further refactoring that 'll address this issue.

Reply:
Actually I can copy a check for now and refactor/deduplicate it later.

Reply:
Done.